src/test/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperationTest.java

```diff
@@ -0,0 +1,295 @@
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@QuarkusTest
+@TestProfile(NoGlobalResourcesTestProfile.Impl.class)
+public class CreateCollectionOperationTest extends OperationTestBase {
+
+  private CommandContext COMMAND_CONTEXT;
+
+  @Inject DatabaseLimitsConfig databaseLimitsConfig;
+
+  @Inject ObjectMapper objectMapper;
+
+  @PostConstruct
+  public void init() {
+    COMMAND_CONTEXT =
+        new CommandContext(
+            KEYSPACE_NAME, COLLECTION_NAME, "testCommand", jsonProcessingMetricsReporter);
+  }
+
+  @Nested
+  class Execute {
+
+    private final ColumnDefinitions RESULT_COLUMNS =
+        buildColumnDefs(OperationTestBase.TestColumn.ofBoolean("[applied]"));
+
+    @Test
+    public void createCollection() {
+      List<Row> deleteRows =
+          Arrays.asList(new MockRow(RESULT_COLUMNS, 0, Arrays.asList(byteBufferFrom(true))));
+
+      AsyncResultSet deleteResults = new MockAsyncResultSet(RESULT_COLUMNS, deleteRows, null);
+      final AtomicInteger schemaCounter = new AtomicInteger();
+      QueryExecutor queryExecutor = mock(QueryExecutor.class);
+      when(queryExecutor.executeCreateSchemaChange(any()))
+          .then(
+              invocation -> {
+                schemaCounter.incrementAndGet();
+                return Uni.createFrom().item(deleteResults);
+              });
+
+      CQLSessionCache sessionCache = mock(CQLSessionCache.class);
+      CqlSession session = mock(CqlSession.class);
+      when(sessionCache.getSession()).thenReturn(session);
+      Metadata metadata = mock(Metadata.class);
+      when(session.getMetadata()).thenReturn(metadata);
+      Map<CqlIdentifier, KeyspaceMetadata> allKeyspaces = new HashMap<>();
+      DefaultKeyspaceMetadata keyspaceMetadata =
+          new DefaultKeyspaceMetadata(
+              CqlIdentifier.fromInternal(KEYSPACE_NAME),
+              false,
+              false,
+              new HashMap<>(),
+              new HashMap<>(),
+              new HashMap<>(),
+              new HashMap<>(),
+              new HashMap<>(),
+              new HashMap<>());
+      allKeyspaces.put(CqlIdentifier.fromInternal(KEYSPACE_NAME), keyspaceMetadata);
+      when(metadata.getKeyspaces()).thenReturn(allKeyspaces);
+
+      CreateCollectionOperation operation =
+          CreateCollectionOperation.withoutVectorSearch(
+              COMMAND_CONTEXT,
+              databaseLimitsConfig,
+              objectMapper,
+              sessionCache,
+              COLLECTION_NAME,
+              "",
+              10,
+              false,
+              false);
+
+      Supplier<CommandResult> execute =
+          operation
+              .execute(queryExecutor)
+              .subscribe()
+              .withSubscriber(UniAssertSubscriber.create())
+              .awaitItem()
+              .getItem();
+      assertThat(schemaCounter.get()).isEqualTo(9);
```

Comment:
Could you add a comment explaining where 9 comes from? (createTable + 8 indexes (including vector)? or something)

Reply:
Excellent!